### PR TITLE
Add keyboard: 7c8/Framework

### PR DESCRIPTION
**Pull request by stevennguyen**
## Description

Add support for 7c8 Framework, an ortholinear keyboard with a rotary encoder.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/7c8/framework/config.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "config_common.h"
+
+#define VENDOR_ID 0x07c8
```
Comment (drashna) [inline keyboards/7c8/framework/config.h:20]: This vendor ID is in use, and should be changed. 

https://www.the-sz.com/products/usbid/index.php?v=0x07c8&p=&n=

File: keyboards/7c8/framework/config.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#define PRODUCT_ID 0x0001
+#define DEVICE_VER 0x0000
+#define MANUFACTURER 7c8
+#define PRODUCT Framework
+#define DESCRIPTION 5x12 ortholinear with a knob
```
Comment (drashna) [inline keyboards/7c8/framework/config.h:25]: This is no longer needed. 
```suggestion
```

File: keyboards/7c8/framework/config.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#define MATRIX_COLS 6
+
+#define MATRIX_ROW_PINS { B0, B1, D7, B2, D6, B3, D5, B4, D4, B5 }
+#define MATRIX_COL_PINS { C0, C1, C2, C3, C4, C5 }
+#define ENCODERS_PAD_A  { D0 }
+#define ENCODERS_PAD_B  { D1 }
+
+#define ENCODER_RESOLUTION 4
+#define TAP_CODE_DELAY 16
+#define UNUSED_PINS
```
Comment (drashna) [inline keyboards/7c8/framework/config.h:37]: Not needed. 
```suggestion
```

File: keyboards/7c8/framework/framework.h [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "quantum.h"
+
+#define ___ KC_NO
+
+#define framework_grid( \
```
Comment (drashna) [inline keyboards/7c8/framework/framework.h:22]:
```suggestion
#define LAYOUT_ortho_5x12( \
```

File: keyboards/7c8/framework/framework.h [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          { K16, K17, K18, K19, K1a, K1b }, \
+          { K20, K21, K22, K23, K24, K25 }, \
+          { K26, K27, K28, K29, K2a, K2b }, \
+          { K30, K31, K32, K33, K34, K35 }, \
+          { K36, K37, K38, K39, K3a, K3b }, \
+          { K40, K41, K42, K43, K44, K45 }, \
+          { K46, K47, K48, K49, K4a, K4b }  \
+        }
+
+#define framework_mit( \
```
Comment (drashna) [inline keyboards/7c8/framework/framework.h:41]:
```suggestion
#define LAYOUT_ortho_5x12_1x2uC( \
```

File: keyboards/7c8/framework/framework.h [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+          { K16, K17, K18, K19, K1a, K1b }, \
+          { K20, K21, K22, K23, K24, K25 }, \
+          { K26, K27, K28, K29, K2a, K2b }, \
+          { K30, K31, K32, K33, K34, K35 }, \
+          { K36, K37, K38, K39, K3a, K3b }, \
+          { K40, K41, K42, K43, K44, K45 }, \
+          { ___, K47, K48, K49, K4a, K4b }  \
+        }
+
+#define framework_2x2u( \
```
Comment (drashna) [inline keyboards/7c8/framework/framework.h:60]:
```suggestion
#define LAYOUT_preonic_2x2u( \
```

File: keyboards/7c8/framework/keymaps/steven/framework.json [hunk 7]
```diff
@@ -0,0 +1,123 @@
+{
```
Comment (drashna) [inline keyboards/7c8/framework/keymaps/steven/framework.json:1]: This looks to be a VIA json, and belongs in the `the-via/keyboards` repo.

File: keyboards/7c8/framework/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    KC_MS_WH_LEFT, KC_MS_WH_RIGHT
+)
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    uint8_t layer = biton32(layer_state);
```
Comment (drashna) [inline keyboards/7c8/framework/keymaps/via/keymap.c:82]:
```suggestion
    uint8_t layer = get_highest_layer(layer_state);
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/7c8/framework/config.h [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "config_common.h"
+
+#define VENDOR_ID 0x77c8
```
Comment (drashna) [inline keyboards/7c8/framework/config.h:20]:
```suggestion
#define VENDOR_ID 0x77C8
```

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**